### PR TITLE
agent: enhance tests of execute_hook

### PR DESCRIPTION
**Pull request by Tim-Zhang**
Use which to find the full path of exe before run execute_hook
to avoid error: 'No such file or directory'

Fixes: #2172

Signed-off-by: Tim Zhang <tim@hyper.sh>

### CODE REVIEW

File: src/agent/rustjail/src/container.rs [hunk 1]
```diff
@@ -1554,19 +1554,35 @@ mod tests {
...
+    async fn which(cmd: &str) -> String {
+        let output: std::process::Output = Command::new("which")
+            .arg(cmd)
+            .output()
+            .await
+            .expect("which command failed to run");
+
+        match String::from_utf8(output.stdout) {
+            Ok(v) => v.trim_end_matches('\n').to_string(),
+            Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
```
Comment (liubin) [inline src/agent/rustjail/src/container.rs:1574]: should avoid `panic` ?

Comment (Tim-Zhang) [reply src/agent/rustjail/src/container.rs:1574]: panic is welcome in the unit test

**Review by mxpv — APPROVED** [review]

**Review by liubin — APPROVED** [review]

File: src/agent/rustjail/src/container.rs [hunk 2]
```diff
@@ -1554,19 +1554,35 @@ mod tests {
...
+    use tokio::process::Command;
 
     macro_rules! sl {
         () => {
             slog_scope::logger()
         };
     }
 
+    async fn which(cmd: &str) -> String {
+        let output: std::process::Output = Command::new("which")
```
Comment (jodh-intel) [inline src/agent/rustjail/src/container.rs:1566]: I wonder if we should be using https://crates.io/crates/which to avoid relying on an external command.

**Pull request merged**